docs_src/design/legacy-requirements/device-service.md

```diff
@@ -0,0 +1,153 @@
+# Device SDK Required Functionality
+
+## Overview
+
+This document sets out the required functionality of a Device SDK other
+than the implementation of its REST API (see [ADR 0011](../adr/device-service/0011-DeviceService-Rest-API.md))
+and the Dynamic Discovery mechanism (see [Discovery](../legacy-design/device-service/discovery.md)).
+
+This functionality is categorised into three areas - actions required at
+startup, configuration options to be supported, and support for push-style
+event generation.
+
+## Startup
+
+When the device service is started, in addition to any actions required to
+support functionality defined elsewhere, the SDK must:
+
+* Register the device service in metadata
+* Provide initialization information to the service implementation
+
+### Registration
+
+The core-metadata service maintains an extent of device service registrations
+so that it may route requests relating to particular devices to the correct
+device service. The SDK should create or update its record appropriately.
+Device service registrations contain the following fields:
+
+* `Name` - the name of the device service
+* `Description` - an optional brief description of the service
+* `Labels` - optional string labels
+* `BaseAddress` - URL of the base of the service's REST API
+
+The default device service `Name` is to be hardcoded into every device service
+implementation. A suffix may be added to this name at runtime by means of
+commandline option or environment variable.
+
+The `Description` and `Labels` are configured in the `[Service]` section of the
+device service configuration.
+
+`BaseAddress` may be constructed using the `[Service]/Host` and `[Service]/Port`
+entries in the device service configuration.
+
+### Initialization
+
+During startup the SDK must supply to the implementation that part of the
+service configuration which is specific to the implementation. This
+configuration is held in the `Driver` section of the configuration file or
+registry.
+
+The SDK must also supply a Logging client at this stage.
```

Comment:
Is this true anymore with logging optional?

Reply:
Perhaps 'facility' would be better than 'client'? The facility should emit logs locally by default but should instead use the optional logging service if so configured.